app/controllers/tasks/import/dwca/psu_import_controller.rb

```diff
@@ -8,19 +8,20 @@ def index
   end
 
   # POST
-  def preview_psu_import
+  def do_psu_import
```

Comment:
There is an isolation of the DWCA part, and the PSU preamble (the controller is, after all, the PSU import controller) The actual batch-load is called from the PSU controller, and is "basic DWCA" (with one exception). There are some PSU-specific objects built in the controller method pre_load.
The exception to which I referred (above) is that the PSU data includes the 'PSUC_FEM' preamble for catalogNumber, which I have used as a namespace, and have isolated from actual catalog number, as per the 'catalogNumber' section of the design document (https://docs.google.com/document/d/18iTlim9lLQXXTdkNUJGXP1WIPcRbYNfkz8dkLUq1YFw/edit)
This one bit of PDU-specific code is in the make_ident section.
In addition, only the columns specific to the PSU data are processed in the current version, although adding additional DWCA-defined (http://rs.tdwg.org/dwc/terms/#dcindex) terms should be simple enough.